tests/test_admin.py

```diff
@@ -131,12 +132,63 @@ def test_get_queryset_filters_by_content_id(self):
         )
 
 
+class MenuItemAdminVersionLocked(CMSTestCase, UsefulAssertsMixin):
+
+    def setUp(self):
+        # who is author..
+        self.menu_content = factories.MenuContentWithVersionFactory(version__state=DRAFT)
+        self.item = factories.ChildMenuItemFactory(parent=self.menu_content.root)
+
+        self.change_url = reverse(
+            "admin:djangocms_navigation_menuitem_change",
+            kwargs={"menu_content_id": self.menu_content.pk, "object_id": self.item.pk}
+        )
+        self.client.force_login(self.get_superuser())
+
+        # moving a node
+        menu_content = factories.MenuContentWithVersionFactory()
+        child = factories.ChildMenuItemFactory(parent=menu_content.root)
+        child_of_child = factories.ChildMenuItemFactory(parent=child)
+        self.move_url = reverse(
+            "admin:djangocms_navigation_menuitem_move_node", args=(menu_content.id,)
+        )
+        self.data = {
+            "node_id": child_of_child.pk,
+            "sibling_id": menu_content.root.pk,
+            "as_child": 1,
+        }
+
+    def test_visit_change_view_when_node_is_version_locked_fails(self):
+        """It fails as super user is not the person who created the version"""
+        response = self.client.get(self.change_url)
+        msg = list(get_messages(response.wsgi_request))[0]
+
+        self.assertEquals(msg.message, "You don't have permission to edit or it is locked")
+        self.assertEquals(response.status_code, 302)
+
+    def test_moving_node_that_is_version_locked_fails(self):
+        response = self.client.post(self.move_url, data=self.data)
+        content = response.content.decode('utf-8')
+        msg = "The item is currently locked or you don't have permission to change it"
+
+        self.assertEquals(response.status_code, 400)
+        self.assertEquals(content, msg)
+
+    @patch('djangocms_navigation.admin.using_version_lock', False)
+    def test_moving_node_version_lock_not_installed_works_without_error(self):
+        response = self.client.post(self.move_url, data=self.data)
+
+        self.assertEquals(response.status_code, 200)
+
+
 class MenuItemAdminChangeViewTestCase(CMSTestCase, UsefulAssertsMixin):
     def setUp(self):
         self.client.force_login(self.get_superuser())
 
     def test_menuitem_change_view(self):
```

Comment:
We should fix this and describe what's being tested in this test. 

"""The change view does not show draft versions"""

This test is a perfect example of why the docstring should always be in a test, I had to work out what the test was doing by reading code. That should not be the case, imagine if 10's of tests were failing and you had to work out if the test needed reworking or the code that it checks. If you know "why" something is being tested then you can decide easily which one to change.

Reply:
Fixed.